jaseci_core/jaseci/svc/common.py

```diff
@@ -125,12 +137,253 @@ def __init__(self):
+            self.patch_apis[api](name=name, body=conf)
+        else:
+            self.patch_apis[api](name=name, namespace=namespace, body=conf)
+
+    def read(self, api: str, name: str, namespace: str = None):
+        if api.startswith("ClusterRole"):
+            return self.read_apis[api](name=name)
+        else:
+            return self.read_apis[api](name=name, namespace=namespace)
+
+    def delete(self, api: str, name: str, namespace: str = None):
+        if api.startswith("ClusterRole"):
+            return self.delete_apis[api](name=name)
+        else:
+            return self.delete_apis[api](name=name, namespace=namespace)
+
+    def is_running(self, name: str, namespace: str):
+        try:
+            return (
+                self.core.list_namespaced_pod(
+                    namespace=namespace, label_selector=f"pod={name}"
+                )
+                .items[0]
+                .status.phase
+                == "Running"
+            )
+        except Exception:
+            return False
+
+    def defaults(self):
+        self.create_apis = {
+            "Service": self.core.create_namespaced_service,
+            "Deployment": self.api.create_namespaced_deployment,
+            "ConfigMap": self.core.create_namespaced_config_map,
+            "ServiceAccount": self.core.create_namespaced_service_account,
+            "ClusterRole": self.auth.create_cluster_role,
+            "ClusterRoleBinding": self.auth.create_cluster_role_binding,
+            "Secret": self.core.create_namespaced_secret,
+            "PersistentVolumeClaim": (
+                self.core.create_namespaced_persistent_volume_claim
+            ),
+            "DaemonSet": self.api.create_namespaced_daemon_set,
+        }
+        self.patch_apis = {
+            "Service": self.core.patch_namespaced_service,
+            "Deployment": self.api.patch_namespaced_deployment,
+            "ConfigMap": self.core.patch_namespaced_config_map,
+            "ServiceAccount": self.core.patch_namespaced_service_account,
+            "ClusterRole": self.auth.patch_cluster_role,
+            "ClusterRoleBinding": self.auth.patch_cluster_role_binding,
+            "Secret": self.core.patch_namespaced_secret,
+            "PersistentVolumeClaim": (
+                self.core.patch_namespaced_persistent_volume_claim
+            ),
+            "DaemonSet": self.api.patch_namespaced_daemon_set,
+        }
+        self.delete_apis = {
+            "Service": self.core.delete_namespaced_service,
+            "Deployment": self.api.delete_namespaced_deployment,
+            "ConfigMap": self.core.delete_namespaced_config_map,
+            "ServiceAccount": self.core.delete_namespaced_service_account,
+            "ClusterRole": self.auth.delete_cluster_role,
+            "ClusterRoleBinding": self.auth.delete_cluster_role_binding,
+            "Secret": self.core.delete_namespaced_secret,
+            "PersistentVolumeClaim": (
```

Comment:
I think we should be thoughtful on the delete behavior of JSORC. Some delete can be destructive and undesirable. I added deletes in my PR too and I didn't include PVC. We should discuss this with @marsninja to get his thoughts.

Reply:
@amadolid As we discussed, let's remove the ability to delete PVC from JSORC.